etc/proto/pachgen/transaction_client.go

```diff
@@ -0,0 +1,113 @@
+//nolint:wrapcheck
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"path"
+	"strings"
+	"text/template"
+
+	descriptor "github.com/golang/protobuf/protoc-gen-go/descriptor"
+	plugin "github.com/golang/protobuf/protoc-gen-go/plugin"
+)
+
+func init() {
+	CodeGenerators = append(CodeGenerators, &TransactionClientGenerator{})
+}
+
+type TransactionClientGenerator struct {
+	Protos []*descriptor.FileDescriptorProto
+}
+
+var funcs = map[string]interface{}{
+	"importPath": func(proto *descriptor.FileDescriptorProto) string {
+		// if *proto.Package == "taskapi" {
+		// 	return `taskapi "github.com/pachyderm/pachyderm/v2/src/task"`
+		// }
+		packageName := *proto.Package
+		if packageName[len(packageName)-3:] == "_v2" {
+			packageName = packageName[:len(packageName)-3]
+		}
+		return fmt.Sprintf(`%s "github.com/pachyderm/pachyderm/v2/src/%s"`, packageName, path.Dir(*proto.Name))
+	},
+	"shouldImport": func(proto *descriptor.FileDescriptorProto) bool {
+		for _, service := range proto.Service {
+			if *service.Name == "API" || *service.Name == "Debug" {
+				return true
+			}
+		}
+		// special case because taskapi has no services defined
+		return *proto.Package == "taskapi"
+	},
+	"isAPI": func(proto *descriptor.ServiceDescriptorProto) bool {
+		return *proto.Name == "API" || *proto.Name == "Debug"
+	},
+	"isClientStreaming": func(proto *descriptor.MethodDescriptorProto) bool {
+		return proto.ClientStreaming != nil && *proto.ClientStreaming
```

Comment:
since these are protos, could just use `GetClientStreaming()`, though I guess we don't do that a lot in our codebase